### ISSUE TITLE
Incorrect exception message for resourceType passed as empty in FhirJsonParser

### ISSUE THREAD

**Issue by rotodd** [FirelyTeam/fhir-net-api#1397]
We merged release [1.7.0](https://github.com/FirelyTeam/fhir-net-api/releases) in [PR1049](https://github.com/microsoft/fhir-server/pull/1049) and are still seeing the same issue detailed in #1304.

**Comment by marcovisserFurore** [comment]
Ai, that should have happened. We will investigate this as soon as possible

**Comment by marcovisserFurore** [comment]
Hi @rotodd, I investigated this issue and I created the following unit test for it:

```
       [TestMethod]
        public void Issue_1304()
        {
            var json = "{\"resourceType\": \"\",  \"id\": \"example\" }";

            var sr = new StringReader(json);
            var jsonreader = new JsonTextReader(sr);
            var parser = new FhirJsonParser();

            var exceptionThrown = false;

            try
            {
                _ = parser.Parse(jsonreader);
            }
            catch (InvalidOperationException ex)
            {
                exceptionThrown = true;
                Assert.IsTrue(ex.Message.Contains("Root object has no type indication (resourceType) and therefore cannot be used to construct an FhirJsonNode."));
            }
            catch (Exception)
            {
                Assert.Fail("Other exceptions are not expected");
            }

            Assert.IsTrue(exceptionThrown, "InvalidOperationException was expected.");
        }
```
This unittest does succeed. It throws now an InvalidOperationException with the message that the resourceType is missing. In the past a FormatException was thrown with the message `Cannot locate type information for type`.

What kind of exception do you receive in your code with fhir-net-api 1.7?

**Comment by marcovisserFurore** [comment]
@rotodd : can you confirm that the issue has been solved? If yes, I will close this issue.

**Comment by marcovisserFurore** [comment]
I close this issue, because there is no response and the unittest proves that the issue has been solved in 1.7
